### PR TITLE
[MM-54497]Fix updatePost props

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
We were using the wrong variable on two different places in the UpdatePost function, causing some issues.

This PR deals with the changes regarding the props, and add tests.

#### Ticket Link
Fix https://mattermost.atlassian.net/browse/MM-54497

#### Release Note
```release-note
Fix post props on update to proper see channel links
```

### CODE REVIEW

File: server/channels/app/post.go [hunk 1]
```diff
@@ -688,7 +688,7 @@ func (a *App) UpdatePost(c *request.Context, post *model.Post, safeUpdate bool)
 		newPost.EditAt = model.GetMillis()
 	}
 
-	if err = a.FillInPostProps(c, post, nil); err != nil {
+	if err = a.FillInPostProps(c, newPost, nil); err != nil {
```
Comment (crspeller) [inline server/channels/app/post.go:691]: Maybe we should rename "post" to prevent this from happening again.

**Review by crspeller — CHANGES_REQUESTED** [review]

**Review by marianunez — APPROVED** [review]

**Comment by marianunez** [comment]
@larkox can you help with QA Tests on how this issue was showing up before to help validate the fix?

**Comment by larkox** [comment]
@yasserfaraazkhan Here are the QA steps.
User experience based testing:
- Make a post with a link to an open channel you are not a member of (e.g. `~channel-name`)
- Refresh the page
- See how the channel is properly linked and show the channel display name (e.g. `~Channel Name`)
- Update the post and add the name of a different channel (e.g. `~channel-name-2`)
- Refresh the page
  - Expected: The channel is linked and show the channel display name
  - Before the fix: The channel shows "raw" (`~channel-name` without a link format)

Technical testing:
- Make a post with a link to an open channel you are not a member of (e.g. `~channel-name`)
- Refresh the page
- The props of the post (that can be found in the unreads network request for this channel) contain the mention for this channel, stating the channel display name.
- Update the post and add the name of a different channel (e.g. `~channel-name-2`)
- Refresh the page
  - Expected: The props of the post (that can be found in the unreads network request for this channel) contain the mention for the new channel, stating the channel display name.
  - Before the fix: The props of the post contains the same information as before the update.

**Review by crspeller — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
Thank you for sharing the steps @larkox 
I've tried those and renaming channel, editing posts, in center, RHS, thread section.

**Pull request merged**